### ISSUE TITLE
[Bug] Logger strings should be % not f-string

### ISSUE THREAD

**Issue by gaby** [asottile/pyupgrade#702]
While recently using pyupgrade to format all the codebase for a big project, I noticed that pyupgrades uses fstrings for loggers which goes against python formatting/optimization rules:

When using pylint to lint all the code it raises W1203 when using fstrings in logging calls: https://pycodequ.al/docs/pylint-messages/w1203-logging-fstring-interpolation.html

Source:
- https://docs.python.org/3/howto/logging.html#optimization
- https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles

Proposed solutions:
- One solution would be to make pyupgrade use % for logging calls by default
- Another solution would be to add a flag to specify that you want pyupgrade to use % instead of f-strings for loggers.

**Comment by asottile** [comment]
you're wrong -- look closer

**Comment by gaby** [comment]
> you're wrong -- look closer

LOL, no wonder there's barely any open issues.

**Comment by yarikoptic** [comment]
just decided to share my particular use case, not necessarily generalizable -- so others might avoid wasting their time.

**TL;DR:** you might be mislead by your eyes and pylint - check if you aren't using `%`-interpolation already instead of passing as args.

**Long version:** I was not sure how to digest above, so I had to investigate our case:  in a trial of using pyupgrade in https://github.com/datalad/datalad/pull/7462 linters started complain since some (not all!) logging invocations were changed to use `.format` , e.g. [here](https://github.com/datalad/datalad/pull/7462/files#diff-dfb40ed9699ae49026787b3f974bfbb3b1fa570186244e443c22d2eae6ded013L73)

```diff
-       lgr.warning("Linking of %s failed (%s), copying file" % (src, e))
+       lgr.warning("Linking of {} failed ({}), copying file".format(src, e))
```

leading to pylint:

```shell
lint: commands[1]> pylint -d all -e W1202 datalad setup.py
************* Module datalad.customremotes.archives
datalad/customremotes/archives.py:73:20: W1202: Use lazy % or % formatting in logging functions (logging-format-interpolation)
************* Module datalad.core.local.resulthooks
datalad/core/local/resulthooks.py:54:[16](https://github.com/datalad/datalad/actions/runs/5693145835/job/15431580675#step:5:17): W1[20](https://github.com/datalad/datalad/actions/runs/5693145835/job/15431580675#step:5:21)2: Use lazy % or % formatting in logging functions (logging-format-interpolation)
************* Module datalad.support.network
datalad/support/network.py:247:24: W1202: Use lazy % or % formatting in logging functions (logging-format-interpolation)
```

looking at those 3 cases I somewhat slowly realized that it was us who used `%` whenever we should have passed it as args to the invocation!, i.e. now I proposed a [PR](https://github.com/datalad/datalad/pull/7475) with

```patch
commit c2bcfe5d8229eaf2f810b5d82a19ad2d52440ca3
Author: Yaroslav Halchenko <debian@onerussian.com>
Date:   Fri Aug 11 11:19:03 2023 -0400

    Replace %-interpolation in log msgs to passing as arguments
    
    This would be "more correct" although, as checked by going through logging code,
    as safe as before.  Detected those through trying pyupgrade which converted
    those to .format() invocations which triggered pylint to start complaining
    about
    
     W1202: Use lazy % or % formatting in logging functions (logging-format-interpolation)
    
    which is still there.  Some elderly issue
    
     https://github.com/pylint-dev/pylint/issues/2395
    
    has some extended discussion on either to keep that warning default or not.
    I think, since it does delay possible "str"-ification of passed args, it is a
    good thing since ideed might have notable (if str() is taking some time for some
    specific types) performance benefits.  But it also points to the need of having
    a CI run (like we do on travis) which has verbose logging enabled to smoke test
    all those code paths.

diff --git a/datalad/core/local/resulthooks.py b/datalad/core/local/resulthooks.py
index 65a7bfeed..5e1aaeb8a 100644
--- a/datalad/core/local/resulthooks.py
+++ b/datalad/core/local/resulthooks.py
@@ -51,8 +51,8 @@ def get_jsonhooks_from_config(cfg):
         )
         if not call:
             lgr.warning(
-                'Incomplete result hook configuration %s in %s' % (
-                    hook_basevar, cfg))
+                'Incomplete result hook configuration %s in %s',
+                hook_basevar, cfg)
             continue
         # split command from any args
         call = call.split(maxsplit=1)
diff --git a/datalad/customremotes/archives.py b/datalad/customremotes/archives.py
index 9a78292c7..2ec905dc6 100644
--- a/datalad/customremotes/archives.py
+++ b/datalad/customremotes/archives.py
@@ -70,7 +70,7 @@ def link_file_load(src, dst, dry_run=False):
         # (e.g. Windows) will not cover scenarios where a particular
         # filesystem simply does not implement it on an otherwise
         # sane platform (e.g. exfat on Linux)
-        lgr.warning("Linking of %s failed (%s), copying file" % (src, e))
+        lgr.warning("Linking of %s failed (%s), copying file", src, e)
         shutil.copyfile(src_realpath, dst)
         shutil.copystat(src_realpath, dst)
     else:
diff --git a/datalad/support/network.py b/datalad/support/network.py
index 77db0b9fb..0ce470ea0 100644
--- a/datalad/support/network.py
+++ b/datalad/support/network.py
@@ -244,7 +244,7 @@ def retry_urlopen(url, retries=3):
         try:
             return __urlopen_requests(url)
         except URLError as e:
-            lgr.warning("Received exception while reading %s: %s" % (url, e))
+            lgr.warning("Received exception while reading %s: %s", url, e)
             if t == retries - 1:
                 # if we have reached allowed number of retries -- reraise
                 raise
```

to make it "right".  So, to some degree it is pylint to blame that it does not complain about using `%` interpolation for the argument but complains about `.format` -- IMHO cases are identical for all noted in various discussions means and purposes really.
